sstables/sstables.cc

```diff
@@ -2935,7 +2935,7 @@ sstable::sstable(schema_ptr schema,
     , _schema(std::move(schema))
     , _generation(generation)
     , _state(state)
-    , _storage(make_storage(manager, storage, make_path(table_dir, _state).native()))
+    , _storage(make_storage(manager, storage, std::move(table_dir), _state))
     , _version(v)
```

Comment:
if storage is stateless with regards to the sstable, why don't we reuse it across sstables under the same storage?

Reply:
I don't follow your point. Storage is not stateless, it contains various data which is sstable-specific, e.g. fs storage keeps full path to where the files really are

Reply:
actually we're pushing sstable state into storage state. I think storage here is more confined to the context of a specific sstable, as fs storage is even holding the temporary dir when writing a new sstable.

Reply:
I agree that it's not extremely nice that we copy state to storage. For now it's like ... initial state :) and state changes are later propagated with the storage::change_state() call. In fact, pretty much every storage method needs `const sstable&` argument. I think that it's worth initializing storage object with the sstable back-reference so this arg and this "initial_state" thing will go away altogether